src/actions/websocket.js

```diff
@@ -132,37 +115,32 @@ function doReconnect() {
         const currentChannelId = getCurrentChannelId(state);
         const currentUserId = getCurrentUserId(state);
 
-        // We try and get the posts for the current channel as soon as possible
-        if (currentChannelId) {
-            dispatch(markChannelAsRead(currentChannelId));
-            dispatch(loadPostsHelper(currentChannelId));
-        }
+        if (currentTeamId) {
+            const dmPrefs = getDirectShowPreferences(state);
+            const statusesToLoad = {
+                [currentUserId]: true,
+            };
+
+            for (const pref of dmPrefs) {
+                if (pref.value === 'true') {
+                    statusesToLoad[pref.name] = true;
+                }
+            }
 
-        dispatch(getLicenseConfig());
-        dispatch(getClientConfig());
-        dispatch(getMyTeams());
-        dispatch(getMyTeamMembers());
-        dispatch(getMyTeamUnreads());
-        dispatch(getMyPreferences());
-        dispatch(loadProfilesForDirect());
-        dispatch(getTeams());
+            dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(() => {
+                dispatch(loadProfilesForDirect());
+            });
 
-        if (currentTeamId) {
-            dispatch(fetchMyChannelsAndMembers(currentTeamId)).then(({data}) => {
-                if (data && currentChannelId) {
-                    const {channels, members} = data;
-                    const stillMemberOfCurrentChannel = members.find((m) => m.channel_id === currentChannelId);
-
-                    if (!stillMemberOfCurrentChannel) {
-                        const defaultChannel = channels.find((c) => c.team_id === currentTeamId && c.name === General.DEFAULT_CHANNEL);
-                        if (defaultChannel) {
-                            EventEmitter.emit(General.DEFAULT_CHANNEL, defaultChannel.display_name);
-                            selectChannel(defaultChannel.id)(dispatch, getState);
-                        }
-                    }
+            dispatch(getPosts(currentChannelId)).then(({error}) => {
+                // Getting the posts return the user is not authorized, redirecting to the default channel
+                if (error && error.status_code === 403) {
+                    EventEmitter.emit(General.DEFAULT_CHANNEL, currentTeamId);
```

Comment:
If it intentional that you're changing the second argument here from the Town Square's display name to the current team's ID?

Also, I know this is how it worked before, but I feel like we should be emitting an event-specific constant instead of just "town-square" here

Reply:
Yes `teamId` is what we want. I've changed the event to be `SWITCH_TO_DEFAULT_CHANNEL`